### DISCUSSION TITLE
[Question] WebBrowser and Excel integration in Windows Forms

### DISCUSSION THREAD

**Discussion by rallets** [dotnet/winforms#6567] [Q&A] [answered]
* .NET Core Version: 

.NET 6.0 - Visual Studio 2022

* Have you experienced this same bug with .NET Framework?: 

No

**Problem description:**

Hi, I hope I'm not too much Off Topic.
In a .net 4.7.2 Windows Forms project a `WebBrowser` control was available. It's not available anymore in a .Net 6.0 project (but I saw some issues here in github mentioning it, so I'm a bit confused, too).
My goal is to use a WebBrowser control to embed a Microsoft Excel stylesheet (so that Excel results inside the Windows Forms application, to avoid context-switching for the end user). 
I know this was possible in the past, but I was not able to find more info about the new .net core version. Do you have any plans in supporting this kind of scenario, or do you know an alternative? I can't reproduce this scenario with my pc because I have installed a newer version of Office (365, excel 16) that I was not able to make it work. It should be possible with previous version of Excel/Office (that is exactly my case).
Related to this I could think to re-use a UserControl built in .net 4.7.2 using the latest version and Visual Studio 2022.
Another option could be to open a generic exe-application inside a UserControl. 
Could this be something supported from .net 6 & VS 2022?

**Expected behavior:**

I can use a WebBrowser and then embed Excel.

**Minimal repro:**

(https://www.codeproject.com/Articles/15760/How-to-Integrate-Excel-in-a-Windows-Form-Applicati)
requires to tweak some registry configs, based on:
(https://social.technet.microsoft.com/Forums/ie/en-US/1fcabd6b-a2e1-4eee-8422-cd77da93ede3/office-2016-open-excel-files-in-ie-11?forum=excel)

**Comment by RussKie** [comment, marked as answer]
> It's not available anymore in a .Net 6.0 project

It is not correct, the control is still available, https://github.com/dotnet/winforms/blob/main/src/System.Windows.Forms/src/System/Windows/Forms/WebBrowser.cs. It is however built on top of the old IE engine, which may no longer work for many web resources (i.e. some website would plainly refuse to render content in it). [WebView2 control](https://docs.microsoft.com/microsoft-edge/webview2/) is a modern alternative, and generally should be considered instead.

**Reply by rallets** [reply]
Hi @RussKie, thank you for your answer. I'm now using WebBrowser, but for me it's not available in the toolbox, but I can instantiate it directly via code. 
For others interested in the same scenario, at the end I reached my goal, and I can show an Excel 2003 (.xls) file inside it, but only in machine with Office 2003 installed (no regedit tweaks needed), instead with Office 365 it opens in a new window.
I used the same code from the link, just with some small adjustments on namespaces and I needed to include the Excel interop dll within the project.
